src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js

```diff
@@ -0,0 +1,303 @@
+        id: 'loanable',
+        name: 'loanable',
+        type: 'checkbox',
+      };
+
+      expect(Field).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.loanable)).toBeInTheDocument();
+    });
+  });
+
+  describe('Render when "isLoanable" returns true', () => {
+    let container;
+
+    beforeEach(() => {
+      container = renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isLoanable: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('Should render loan profile dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        id: 'input_loan_profile',
+        name: 'loansPolicy.profileId',
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(2, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.loanProfile)).toBeInTheDocument();
+    });
+
+    it('Should render due date dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        name: 'loansPolicy.closedLibraryDueDateManagementId',
+        children: {
+          options: longTermLoansOptions,
+          id: messageIds.pleaseSelect,
+        }
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(3, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.dueDate)).toBeInTheDocument();
+    });
+
+    it('Should render item limit input correctly', () => {
+      const expectedProps = {
+        component: TextField,
+        name: 'loansPolicy.itemLimit',
+        id: 'input_item_limit',
+        type: 'number',
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(4, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.itemLimit)).toBeInTheDocument();
+    });
+
+    it('"Period" component should get correct properties under "Grace period" section', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.gracePeriod,
+        inputValuePath: 'loansPolicy.gracePeriod.duration',
+        selectValuePath: 'loansPolicy.gracePeriod.intervalId',
+        changeFormValue: initialProps.change,
+      };
+
+      expect(Period).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+  });
+
+  describe('Render when "isProfileRolling" returns true', () => {
+    let container;
+
+    beforeEach(() => {
+      container = renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isProfileRolling: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('"Period" component should get correct properties under "Loan period" section', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.loanPeriod,
+        inputValuePath: 'loansPolicy.period.duration',
+        selectValuePath: 'loansPolicy.period.intervalId',
+        changeFormValue: initialProps.change,
```

Comment:
The same with `intervalPeriods` as abowe.
Relates to the whole component, I believe.

Reply:
fixed